tsl/src/remote/connection.c

```diff
@@ -1468,6 +1486,45 @@ add_userinfo_to_server_options(ForeignServer *server, Oid user_id)
 	return options;
 }
 
+char *
+remote_connection_get_connstr(const char *node_name)
+{
+	ForeignServer *server;
+	List *connection_options;
+	const char **keywords;
+	const char **values;
+	StringInfoData buf;
+	int i;
+
+	server = data_node_get_foreign_server(node_name, ACL_NO_CHECK, false, false);
+	connection_options = add_userinfo_to_server_options(server, GetUserId());
+	setup_full_connection_options(connection_options, &keywords, &values);
+
+	/* Cycle through the options and create the connection string */
+	initStringInfo(&buf);
+	i = 0;
+	while (keywords[i] != NULL)
+	{
+		int value_len = strlen(values[i]);
+
+		appendStringInfo(&buf, " %s=", keywords[i]);
+
+		/* ensure there is enough space reserved for escaped value */
+		enlargeStringInfo(&buf, value_len * 2 + 1);
+
+		/* escape value */
+		buf.len += PQescapeString(buf.data + buf.len, values[i], value_len);
```

Comment:
i think the correct function to use here is appendConnStrVal and then call PQescapeString on the whole string you pass as conninfo into CREATE SUBSCRIPTION, you are not treating the individual key-value pairs of the connection string correctly here, values have to be wrapped in quotes if they contain spaces.

Reply:
It seems to be `appendConnStrVal` is part of the libpq source which is not part of imported PostgreSQL include directory we are using and sits in the `interfaces/libpq` directory. It is not clear how we should add this include path unless we always provide `PG_SOURCE_DIR` which we don't. Looks like it is easier to use `PQescapeString()` in both cases here